### DISCUSSION TITLE
[Feature] Relationship/closeness graph between people

### DISCUSSION THREAD

**Discussion by raminrahimzada** [immich-app/immich#14744] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I don't have so much knowledge about the internals so please give me some hints if this is already possible by using db
Just wanted to create a graph view of people from recognized faces using this strategy :

Let each person from recognized faces be a `node`
If `Person1` has `p1` images,
 `Person2` has `p2` images,
 `Person1` and `Person2`  have `n>0` mutual pictures, we say there is an `edge` between `Person1` and `Person2` and the weight of edge is `F(p1,p2,n)`  
We should select this function so that more mutual pictures should result higher weight (weights are between 0 and 1) 
For example `F(p1,p2,n)=n/(p1+p2)`

So after populating all the edges&weights we will be able to answer these type of questions :
* Who knows `Person1`
* Who is much closer to `Person1` [`PersonA` or `PersonB`] ?

I found an example image from web (despite it does not show the weights)
<img src="https://i.sstatic.net/HHt4x.png" alt="https://i.sstatic.net/HHt4x.png"/>

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This is a cool idea, but not really in scope for Immich.
